.github/workflows/build.yml

```diff
@@ -150,8 +150,10 @@ jobs:
           - name: linux-O0
             os: ubuntu-latest
             config_arg: CFLAGS='-O0'
-          - name: macos
-            os: macos-latest
+          - name: macos-x86_64
+            os: macos-14-large
```

Comment:
```suggestion
            os: macos-latest
```

Reply:
from the blog post:

> The macOS 12 runner image will remain latest until migration of the latest YAML workflow label to macOS 14 in Q2 FY24 (April – June 2024).

Reply:
and 

> While macOS 13 is now generally available under the macos-13 label, this image will not be migrated to latest.

perhaps this should be 13 so we make sure to stay on x86_64?

Reply:
`macos-14-large` seems to be x86_64 in my testing while `macos-14` is ARM64.
See the Checks performed on https://github.com/tarides/runtime_events_tools/pull/42.

Reply:
oh wow, that's confusing. indeed, from https://github.com/actions/runner-images/tree/main?tab=readme-ov-file#available-images you're right

<img width="650" alt="image" src="https://github.com/ocaml/ocaml/assets/661909/cd5a65f9-3640-4ea4-92c0-0934b43160ea">

Reply:
The real joy is in debugging the yaml 😬 

I don't personally see much point in keeping macOS 12 or 13 around in the build matrix, but others might disagree.

Reply:
Good news is `macos-13` can stay on as an x86_64 build. 
Bad news is `macos-14-large` seems to cost more than the plan present on `ocaml` account in GitHub. Not sure what the cost / plan is, perhaps @avsm might know.